tensorflow/lite/micro/kernels/leaky_relu.cc

```diff
@@ -23,122 +23,129 @@ limitations under the License.
     QuantizeMultiplier(alpha_multiplier, &data->output_multiplier_alpha,
-                       &data->output_shift_alpha);
-    double identity_multiplier = input->params.scale / output->params.scale;
+                       &output_shift_alpha);
+    data->output_shift_alpha = static_cast<int32_t>(output_shift_alpha);
+
+    int output_shift_identity;
+    double identity_multiplier =
+        static_cast<double>(input->params.scale / output->params.scale);
     QuantizeMultiplier(identity_multiplier, &data->output_multiplier_identity,
-                       &data->output_shift_identity);
+                       &output_shift_identity);
+    data->output_shift_identity = static_cast<int32_t>(output_shift_identity);
   }
 
-  if (input->type == kTfLiteInt16 && output->type == kTfLiteInt16) {
-    TF_LITE_ENSURE_EQ(context, input->params.zero_point, 0);
-    TF_LITE_ENSURE_EQ(context, output->params.zero_point, 0);
-  }
+  return kTfLiteOk;
+}
+
+void* LeakyReluInit(TfLiteContext* context, const char* buffer, size_t length) {
+  TFLITE_DCHECK(context->AllocatePersistentBuffer != nullptr);
+  return context->AllocatePersistentBuffer(context, sizeof(LeakyReluOpData));
+}
 
-  return kTfLiteError;
+TfLiteStatus LeakyReluPrepare(TfLiteContext* context, TfLiteNode* node) {
+  return CalculateOpData(context, node);
 }
 
 TfLiteStatus LeakyReluEval(TfLiteContext* context, TfLiteNode* node) {
-  const TfLiteTensor* input;
-  TF_LITE_ENSURE_OK(context, GetInputSafe(context, node, 0, &input));
-  TfLiteTensor* output;
-  TF_LITE_ENSURE_OK(context, GetOutputSafe(context, node, 0, &output));
-  const auto* params =
-      reinterpret_cast<TfLiteLeakyReluParams*>(node->builtin_data);
-  const LeakyReluOpData* data =
-      reinterpret_cast<LeakyReluOpData*>(node->user_data);
+  const TfLiteEvalTensor* input =
+      tflite::micro::GetEvalInput(context, node, kInputTensor);
+  TfLiteEvalTensor* output =
+      tflite::micro::GetEvalOutput(context, node, kOutputTensor);
+  const LeakyReluOpData& data = *static_cast<LeakyReluOpData*>(node->user_data);
 
-  LeakyReluParams op_params;
   switch (input->type) {
     case kTfLiteFloat32: {
+      LeakyReluParams op_params = {};
+      const auto* params =
+          static_cast<TfLiteLeakyReluParams*>(node->builtin_data);
+
       op_params.alpha = params->alpha;
-      reference_ops::LeakyRelu(
-          op_params, GetTensorShape(input), GetTensorData<float>(input),
-          GetTensorShape(output), GetTensorData<float>(output));
-      return kTfLiteOk;
-    } break;
-    case kTfLiteUInt8: {
-      QuantizeLeakyRelu<uint8_t>(input, output, data);
+      reference_ops::LeakyRelu(op_params, tflite::micro::GetTensorShape(input),
+                               tflite::micro::GetTensorData<float>(input),
+                               tflite::micro::GetTensorShape(output),
+                               tflite::micro::GetTensorData<float>(output));
       return kTfLiteOk;
-    } break;
```

Comment:
A bit of a nit-pick, but could you add "break" back in here, and below? I know the return earlier makes it redundant, but if the body of the case statement was changed in the future, we'd end up with an accidental fall through.

Reply:
Fixed.